plugins/dell-dock/fu-plugin-dell-dock.c

```diff
@@ -165,6 +188,16 @@ void
 fu_plugin_device_registered (FuPlugin *plugin, FuDevice *device)
 {
 	/* thunderbolt plugin */
+	if (g_strcmp0 (fu_device_get_plugin (device), "thunderbolt") == 0 &&
+	    fu_device_has_guid (device, "TBT-00d4b071")) {
+		fu_device_inhibit (device, "dell-dock-usb4", "handled by usb");
+		if (fu_device_has_flag(device, FWUPD_DEVICE_FLAG_UPDATABLE)) {
+			g_print("failed to inhibit device %s:%s",
+				fu_device_get_plugin (device),
+				fu_device_get_name (device));
```

Comment:
really?  is this needed?  or just debugging.

Reply:
I'm facing an interesting issue that device somewhat showing different outcome of flags after inhibit.

Reply:
Using `verbose` output I can confirm that `updatable` flag is gone, together with `updatable-hidden` added.  However, without `verbose` it’s back again… And, ultimately `USB4` device from `thunderbolt` plugin still gets the firmware install because it’s in fact `updatable` right after `dell_dock`’s `USB4`.

#fwupdtool get-devices -v
[fwupdtool-get-devices-v-ok.log](https://github.com/fwupd/fwupd/files/6660073/fwupdtool-get-devices-v-ok.log)

#fwupdtool get-devices
[fwupdtool-get-devices-ng.log](https://github.com/fwupd/fwupd/files/6660075/fwupdtool-get-devices-ng.log)

Reply:
Well that's exactly expected behavior - see: https://fwupd.github.io/libfwupdplugin/method.Device.inhibit.html

Reply:
Is this because of the call to `fu_dell_dock_clone_updatable` perhaps afterwards?  Maybe you need to make sure that is only called for TBT3 registered call, not TBT4.

Reply:
Removed `fu_dell_dock_clone_updatable`, afterwards `USB4` from `thunderbolt` plugin has no `updatable` flag but it was anyway prepared, and exercised an update. Trying to find out why.

[no-clone-updatable-ng.log](https://github.com/fwupd/fwupd/files/6660960/no-clone-updatable-ng.log)

Reply:
See if this fixes it:
```
diff --git a/src/fu-install-task.c b/src/fu-install-task.c
index b874e72f..3b25ac3e 100644
--- a/src/fu-install-task.c
+++ b/src/fu-install-task.c
@@ -297,8 +297,7 @@ fu_install_task_check_requirements (FuInstallTask *self,
        }
 
        /* no update abilities */
-       if (!fu_device_has_flag (self->device, FWUPD_DEVICE_FLAG_UPDATABLE) &&
-           !fu_device_has_flag (self->device, FWUPD_DEVICE_FLAG_UPDATABLE_HIDDEN)) {
+       if (!fu_device_has_flag (self->device, FWUPD_DEVICE_FLAG_UPDATABLE) {
                g_set_error (error,
                             FWUPD_ERROR,
                             FWUPD_ERROR_NOT_SUPPORTED,
```

Reply:
that fix is merged to master, rebase on master and try again.

Reply:
Issue is fixed with 4554eaa6, thanks!